lightning/src/ln/onchaintx.rs

```diff
@@ -462,235 +293,71 @@ impl<ChanSigner: ChannelKeys> OnchainTxHandler<ChanSigner> {
-							let feerate = (amount - htlc_tx.output[0].value) * 1000 / htlc_tx.get_weight() as u64;
-							// Timer set to $NEVER given we can't bump tx without anchor outputs
-							log_trace!(logger, "Going to broadcast Local HTLC-{} claiming HTLC output {} from {}...", if preimage.is_some() { "Success" } else { "Timeout" }, outp.vout, outp.txid);
-							return Some((None, feerate, htlc_tx));
-						}
-						return None;
-					},
-					&InputMaterial::Funding { ref funding_redeemscript } => {
-						let signed_tx = self.get_fully_signed_local_tx(funding_redeemscript).unwrap();
-						// Timer set to $NEVER given we can't bump tx without anchor outputs
-						log_trace!(logger, "Going to broadcast Local Transaction {} claiming funding output {} from {}...", signed_tx.txid(), outp.vout, outp.txid);
-						return Some((None, self.local_commitment.as_ref().unwrap().feerate_per_kw, signed_tx));
-					}
-					_ => unreachable!()
-				}
+			if let Some(transaction) = cached_request.content.package_finalize(self, amt, self.destination_script.clone(), &logger) {
+				return Some((None, self.local_commitment.as_ref().unwrap().feerate_per_kw, transaction));
 			}
 		}
 		None
 	}
 
-	pub(super) fn block_connected<B: Deref, F: Deref, L: Deref>(&mut self, txn_matched: &[&Transaction], claimable_outpoints: Vec<ClaimRequest>, height: u32, broadcaster: B, fee_estimator: F, logger: L)
+	pub(super) fn block_connected<B: Deref, F: Deref, L: Deref>(&mut self, txn_matched: &[&Transaction], requests: Vec<OnchainRequest>, height: u32, broadcaster: B, fee_estimator: F, logger: L)
 		where B::Target: BroadcasterInterface,
 		      F::Target: FeeEstimator,
 					L::Target: Logger,
 	{
-		log_trace!(logger, "Block at height {} connected with {} claim requests", height, claimable_outpoints.len());
-		let mut new_claims = Vec::new();
-		let mut aggregated_claim = HashMap::new();
-		let mut aggregated_soonest = ::std::u32::MAX;
+		log_trace!(logger, "Block at height {} connected with {} claim requests", height, requests.len());
+		let mut preprocessed_requests = Vec::with_capacity(requests.len());
+		let mut aggregated_request = OnchainRequest::default();
 
 		// Try to aggregate outputs if their timelock expiration isn't imminent (absolute_timelock
 		// <= CLTV_SHARED_CLAIM_BUFFER) and they don't require an immediate nLockTime (aggregable).
-		for req in claimable_outpoints {
+		for req in requests {
 			// Don't claim a outpoint twice that would be bad for privacy and may uselessly lock a CPFP input for a while
-			if let Some(_) = self.claimable_outpoints.get(&req.outpoint) { log_trace!(logger, "Bouncing off outpoint {}:{}, already registered its claiming request", req.outpoint.txid, req.outpoint.vout); } else {
+			if let Some(_) = self.claimable_outpoints.get(req.content.outpoints()[0]) { log_trace!(logger, "Bouncing off outpoint {}:{}, already registered its claiming request", req.content.outpoints()[0].txid, req.content.outpoints()[0].vout); } else {
```

Comment:
you keep accessing the first outpoint, so I'd just put it in a variable for easier legibility